app-backend/batch/src/main/scala/com/azavea/rf/batch/landsat8/ImportLandsat8.scala

```diff
@@ -0,0 +1,274 @@
+    logger.debug(s"Constructed Root URL: $rootUrl")
+    rootUrl
+  }
+
+  protected def s3ObjExists(url: String): Boolean = {
+    try {
+      Source.fromURL(url, "UTF-8")
+      true
+    } catch {
+      case _: java.io.FileNotFoundException => false
+    }
+  }
+
+  protected def sizeFromPath(tifPath: String, landsatId: String): Int = {
+    val path = s"${getLandsatPath(landsatId)}$tifPath"
+    logger.info(s"Getting object size for path: $path")
+    S3.getObject(landsat8Config.bucketName, path).getObjectMetadata.getContentLength.toInt
+  }
+
+  protected def createThumbnails(sceneId: UUID, landsatId: String) = {
+    val path = getLandsatUrl(landsatId)
+    val smallUrl = s"$path${landsatId}_thumb_small.jpg}"
+    val largeUrl = s"$path${landsatId}_thumb_large.jpg}"
+
+    Thumbnail.Identified(
+      id = None,
+      organizationId = UUID.fromString(landsat8Config.organization),
+      thumbnailSize = ThumbnailSize.Small,
+      widthPx = 228,
+      heightPx = 233,
+      sceneId = sceneId,
+      url = smallUrl
+    ) :: Thumbnail.Identified(
+      id = None,
+      organizationId = UUID.fromString(landsat8Config.organization),
+      thumbnailSize = ThumbnailSize.Large,
+      widthPx = 1143,
+      heightPx = 1168,
+      sceneId = sceneId,
+      url = largeUrl
+    ) :: Nil
+  }
+
+
+  @SuppressWarnings(Array("TraversableHead"))
+  protected def csvRowToScene(row: Map[String, String], srcProj: CRS = CRS.fromName("EPSG:4326"), targetProj: CRS = CRS.fromName("EPSG:4326")): Future[Option[Scene.Create]] = Future {
+    val sceneId = UUID.randomUUID()
+    val landsatId = row("sceneID")
+
+    val ll = row("lowerLeftCornerLongitude").toDouble -> row("lowerLeftCornerLatitude").toDouble
+    val lr = row("lowerRightCornerLongitude").toDouble -> row("lowerRightCornerLatitude").toDouble
+    val ul = row("upperLeftCornerLongitude").toDouble -> row("upperLeftCornerLatitude").toDouble
+    val ur = row("upperRightCornerLongitude").toDouble -> row("upperRightCornerLatitude").toDouble
+
+    val srcCoords = ll :: lr :: ul :: ur :: Nil
+
+    val sortedByX = srcCoords.sortBy(_.x)
+    val sortedByY = srcCoords.sortBy(_.y)
+
+    val extent = Extent(
+      xmin = sortedByX.head.x,
+      ymin = sortedByY.head.y,
+      xmax = sortedByX.last.x,
+      ymax = sortedByY.last.y
+    )
+
+    val (transformedCoords, transformedExtent) = {
+      if (srcProj.equals(targetProj)) Polygon(Line(srcCoords :+ srcCoords.head)) -> extent
+      else
+        Polygon(Line(srcCoords.map {
+          _.reproject(srcProj, targetProj)
+        })) -> extent.reproject(srcProj, targetProj)
+    }
+
+    val landsatPath = getLandsatPath(landsatId)
+
+
+    if (!s3ObjExists(s"${landsat8Config.awsLandsatBase}${landsatPath}index.html")) {
+      logger.warn(
+        "AWS and USGS are not always in sync. Try again in several hours.\n" +
+          s"If you believe this message is in error, check ${landsat8Config.awsLandsatBase}${landsatPath} manually."
```

Comment:
A few comments:
 - I think this URL should be the same as the one that is checked, else the warning message is kind of confusing because `http://landsat-pds.s3.amazonaws.com/L8/096/043/LC80960432016080LGN00` (which shows up in the error message) does not exist, but `http://landsat-pds.s3.amazonaws.com/L8/096/043/LC80960432016080LGN00/index.html` does exist -- so constructing the URL once and using it in both places to DRY it out might be better
 - I think there is a typo in the `s3ObjExists` check because the importer fails for every scene -- I believe there should be a `/` before `index.html`
 - I think the method call could be something better -- it _is_ an `s3` object, but it's a URL -- I was half expecting to see `s3://` rather than `http://`. I'm not sure what a better name would be, but this is the only place this is called, so perhaps a better name would be `checkIfSceneExists`?